client/rest.go

```diff
@@ -0,0 +1,259 @@
+package client
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"github.com/FourthState/plasma-mvp-sidechain/plasma"
+	"github.com/FourthState/plasma-mvp-sidechain/store"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethcmn "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/gorilla/mux"
+	"math/big"
+	"net/http"
+)
+
+func RegisterRoutes(ctx context.CLIContext, r *mux.Router) {
+	r.HandleFunc("/balance/{address}", balanceHandler(ctx)).Methods("GET")
+	r.HandleFunc("/info/{address}", infoHandler(ctx)).Methods("GET")
+	r.HandleFunc("/block/{height}", blockHandler(ctx)).Methods("GET")
+	r.HandleFunc("/blocks/{height}", blocksHandler(ctx)).Methods("GET")
+	r.HandleFunc("/tx/{hash}", txHandler(ctx)).Methods("GET")
+	r.HandleFunc("/output/{position}", outputHandler(ctx)).Methods("GET")
+	r.HandleFunc("/submit", submitHandler(ctx)).Methods("POST")
+}
+
+func balanceHandler(ctx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		addr := vars["address"]
+		if !ethcmn.IsHexAddress(addr) {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("address must be an ethereum 20-byte hex string"))
+			return
+		}
+
+		total, err := Balance(ctx, ethcmn.HexToAddress(addr))
+		if err != nil {
+			sdkerr, ok := err.(sdk.Error)
+			if !ok || sdkerr.Code() != store.CodeDNE {
+				w.WriteHeader(http.StatusBadRequest)
+			} else {
+				w.WriteHeader(http.StatusOK)
```

Comment:
I don't think we should write StatusOK if an error occured

Reply:
The reason for this was that if the address of a wallet doesn't exits, it's not necessarily a bad request. Maybe StatusNotFound(403)?

Reply:
its 404 but yea, I think that would be good

Reply:
what would it the error look like for DNE

Reply:
ah yeah, 404. It would be the stringified version of sdk.Error. something link "codespace: <code> code: <code> message: <msg>"

Reply:
Hmm yea, you can make the call on this, I think either 404 or StatusOK are fine solutions

Reply:
I'll change it to StatusNotFound. I think that's is more clear